src/string.cr

```diff
@@ -546,10 +550,20 @@ class String
         when 'x'
           base = 16
           ptr += 1
-        else
+        when 'o'
           base = 8
+          ptr += 1
+        else
+          if leading_zero_is_octal
+            base = 8
+            ptr -= 1
```

Comment:
Why decrement the pointer?

Reply:
After looking closer, I don't think it is needed. It was to cover the case where there's a leading zero, but no `o` right after it. It would handle strings like `0123` and `0abc` (invalid). I think I put it in because there's no prefix character (like b or x), so it needed to back up.

Reply:
Travis [build 15307.1](https://travis-ci.org/crystal-lang/crystal/builds/527505579) failed when this was removed. Looks like it broke parsing of `!!int 0` in YAML. I'll see if there's a way I can address this without decrementing the pointer, unless you're ok with putting it back in.